server/src/models/userProfile.js

```diff
@@ -0,0 +1,48 @@
+const mongoose = require("mongoose");
+
+const userProfileSchema = new mongoose.Schema(
+  {
+    userField: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "user",
+      required: true,
+    },
+    isDoctor: {
```

Comment:
I have created a separate model for Doctor, so you can remove the isDoctor field.